sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableIO.java

```diff
@@ -131,6 +134,32 @@
  *         .withTableId("table"));
  * }</pre>
  *
+ * <p>Optionally, BigtableIO.write() may be configured to emit {@link BigtableWriteResult} elements
+ * after each group of inputs is written to Bigtable. These can be used to then trigger user code
+ * after writes have completed, for example:
+ *
+ * <pre>{@code
+ * // Note, data must have a windowing function set to something other than the global window.
```

Comment:
I am curious about this comment. I understand the logic of it, but I am curious if we can tackle the case of triggering the 'next' part after full writing a PCollection, aka the Batch case. Any hints @jkff ? (I thought this already worked also for that case or maybe I just misread it)

Reply:
Mmm actually after looking at the [Wait javadoc](https://github.com/iemejia/beam/blob/1ad61fd384bcd1edd11086a3cf9d7dddb154d934/sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Wait.java#L68) I have the impression the global window is covered too but it requires the input to be bounded. So maybe this comment is not needed.

`If signal is globally windowed, main input must also be. This typically would be useful only in a batch pipeline, because the global window of an infinite PCollection never closes, so the wait signal will never be ready. `

Reply:
oh yeah, this works for batch...we even use it in batch mode already.  I'm not sure why I added that comment.  I'll remove it.